drivers/include/net/netdev2.h

```diff
@@ -10,15 +10,153 @@
+ * context. With reasonable effort and resource usage, we have no means of
+ * synchronizing the bus access also in interrupt context.
+ *
+ * The solution to this problem as implemented by this interface is **not to
+ * call any function that interacts with a device directly from interrupt
+ * context**. Unfortunately this requires some added complexity for
+ * synchronization efforts between thread and interrupt context to be able to
+ * handle device events (i.e. external interrupts). See section @res sec_events
+ * for more information.
+ *
+ *
+ * @section sec_context Context requirements
+ *
+ * The `netdev` interface expects the network device drivers to run in thread
+ * context (see section above). The interface was however designed in a ways, to
+ * allow more than one device driver to be serviced in the same thread.
+ *
+ * The key design element for `netdev` is, that device drivers implementing this
+ * interface are not able to run stand-alone in a thread, but need some
+ * bootstrapping code. This bootstrapping code can be anything from a simple
+ * msg_receive() loop (as done for the GNRC adaption) to a complete network
+ * stack that works without messaging entirely but is build on function call
+ * interfaces.
+ *
+ *
+ * @section sec_sendrecv Sending and Receiving
+ *
+ * Sending data using the `netdev` interface is straight forward: simply call
+ * the drivers send() function, passing it the data that should be send. The
+ * caller of the send() function (e.g. a network stack) must hereby make sure,
+ * that the data is in the correct format expected by the specific network
+ * device driver. Typically the data needs to contain a pre-filled link layer
+ * header as e.g. an IEEE802.15.4 or Ethernet header.
+ *
+ * Receiving data using the `netdev` interface requires typically four steps:
+ * 1. wait for a NETDEV_EVENT_RX_COMPLETE event
+ * 2. call the recv() function with `buf := NULL` and `len := 0` to get the size
+ *    of the received data
+ * 3. allocate a large enough buffer in some way
+ * 4. call the recv() function a second time, passing the buffer and reading the
+ *    received data into this buffer
+ *
+ * This receive sequence can of course be simplified by skipping steps 2 and 3
+ * when using fixed sized pre-allocated buffers or similar means. *
+ *
+ * @note    The send() and recv() functions **must** never be called from
+ *          interrupt context.
+ *
+ * @section sec_config Device Configuration
+ *
+ * The `netdev` interface covers a wide variety of network devices, which differ
+ * to some extend in their configuration parameters (e.g. radios vs. wired
+ * interfaces, channel selection vs. link status detection). To cover this
+ * variety, `netdev` provides a generic configuration interface by exposing
+ * simple get() and set() functions. These are base on a globally defined and
+ * extendable list of options as defined in @ref netopt.h.
```

Comment:
considering some questions on the mailing list, I propose to **emphasis** `**extendable**` ;-)